.github/workflows/unit-tests.yml

```
@@ -2,80 +1,190 @@
+          # Run tests and capture exit code
+          npm test 2>&1 | tee test-output.log
+          test_exit_code=${PIPESTATUS[0]}
+          
+          end_time=$(date +%s%3N)
+          duration=$((end_time - start_time))
+          
+          # Extract test counts from output
+          total_tests=$(grep -oP '\d+(?= passing)' test-output.log | head -1 || echo "0")
+          passing_tests=$(grep -oP '\d+(?= passing)' test-output.log | head -1 || echo "0")
+          failing_tests=$(grep -oP '\d+(?= failing)' test-output.log | head -1 || echo "0")
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)